src/main/java/uk/gov/hmcts/reform/sscs/robotics/RoboticsJsonMapper.java

```diff
@@ -240,8 +242,14 @@ private static String getCaseCode(SscsCaseData sscsCaseData) {
             return sscsCaseData.getCaseCode();
             // Leave this in for now, whilst we have legacy cases where the case code is not set.
             // This will be an issue for cases where the caseworker tries to regenerate the robotics json. Can remove after a few weeks I suspect.
-        } else if (equalsIgnoreCase("esa", sscsCaseData.getAppeal().getBenefitType().getCode())) {
+        }
+
+        Benefit benefit = Benefit.findBenefitByShortName(sscsCaseData.getAppeal().getBenefitType().getCode());
+
+        if (ESA == benefit) {
             return ESA_CASE_CODE;
+        } else if (CARERS_ALLOWANCE ==  benefit) {
```

Comment:
Don't need this - it was only put in for legacy cases in Aug 2019. Needs to be removed at some point but Hem said we still get some through.